src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -100,7 +100,7 @@ public static bool IsLockFileStillValid(DependencyGraphSpec dgSpec, PackagesLock
 
                             var projectDependency = target.Dependencies.FirstOrDefault(
                                 dep => dep.Type == PackageDependencyType.Project &&
-                                PathUtility.GetStringComparerBasedOnOS().Equals(dep.Id, p2pProjectName));
```

Comment:
What happens if just the comparison is case insensitive? 

The original issue is "resolved", but say next time you generate the lock file, the correspondence between the old and new dependency entry might contain a changed casing?

Reply:
Actually I think it's enough to change just this. 

Effectively by making the comparer always insensitive you are enforcing the windows behavior everywhere. 

I'll do some tests and get back to you.

Reply:
I just ran the tests. I don't really see the impact here. What's the bad thing that happens from having mismatched casing as long as our RestoreLockedMode comparison works fine.

Reply:
I'd suggest just fixing the comparison and not making the lock file unnecessarily weird. 
We'd get a similar complaint from customers when they see the project names are always lowercase but the package ids are not, yet the package ids are truly case insensitive.

Reply:
As discussed, the comparison is already case-insensitive, as you found. The issue is entirely about when something causes the lock file to be changed, the change to the file will be greater than what caused the change. For example, if the project added a new NuGet package, the lock file diff will contain not only the new NuGet package, but also every project name.